torch/_dynamo/output_graph.py

```diff
@@ -175,13 +175,15 @@ def __init__(
         f_globals: Dict[str, Any],
         code_options: Dict[str, Any],
         compiler_fn: CompilerFn,
+        tracing_context: TracingContext,
         root_tx,
     ):
         super(OutputGraph, self).__init__()
 
         self.graph = torch.fx.Graph()
         self.graphargs: List[GraphArg] = []
-        self.guards: Set[Guard] = set()
+        self.tracing_context: TracingContext = tracing_context
+        self.guards: Set[Guard] = self.tracing_context.guards_context.dynamo_guards
```

Comment:
So not going to update checkpointing yet?

Reply:
As this code currently stands it is wrong; checkpointing is still writing to guards and not updating the context

Reply:
Sorry, maybe I misunderstood - but isn't this the same object?

Reply:
The problem is you setup it to alias here, but at 

https://github.com/pytorch/pytorch/blob/c356f1a40d18497b33ec9e7c13ee7afacee0b0d0/torch/_dynamo/output_graph.py#L254

it gets overwritten, destroying the aliasing relationship.

Reply:
Fixed

Reply:
I think I'd prefer this as a `@property`, as it will prevent people from accidentally overwriting it in the future

Reply:
Good suggestion